[FEATURE] add remote loader for runtime plugins

jgbernalp opened the pull request:
# Description

This PR uses the remote plugin loader to load all types of plugins at runtime, based on the installed plugins fetched from `api/v1/plugins`

This will work in conjunction with https://github.com/perses/plugins/pull/16 and https://github.com/perses/plugins/pull/18

Nexucis commented on ui/plugin-runtime/package.json:
```diff
@@ -28,7 +28,8 @@
   "dependencies": {
-    "@module-federation/enhanced": "^0.1.21"
+    "@module-federation/enhanced": "^0.1.21",
+    "@perses-dev/plugin-system": "0.49.0"
```
do you foresee in a long term that `plugin-system` and `plugin-runtime` are merged and there is only one package for the plugin system ?

jgbernalp replied:
yes I started as its own package to avoid breaking the current system, prior an initial release we can merge them

jgbernalp commented on ui/plugin-runtime/src/remotePluginLoader.ts:
```diff
@@ -0,0 +1,60 @@
+
+      const pluginModules = plugins.map((plugin: Record<string, unknown>) => {
+        return {
+          kind: 'PluginModule',
```
true, let me make the changes.

Nexucis commented on internal/api/plugin/plugin.go:
```diff
@@ -17,18 +17,39 @@ import (
+	pluginFileName   = "plugin-modules.json"
+)
+
+type PluginMetadata struct {
```
I propose to put that in `pkg/model/api/v1` and then to use the classic `Kind, Metadata, Spec`

Then in spec, you can have the Display struct and the list of the plugins

jgbernalp replied:
I tried to use the default Metadata but the version there is numeric, and for the case of the plugins it will use semver. Should we change it there? Am I missing something?

Nexucis replied:
ah yeah I forgot it contains a version in number.

So yeah let's create another metadata for the plugin only. But I would still consider to use the classic struct `Kind`, `Metadata`, `Spec` even if `Metadata` is another one.

Something like that

```go
type PluginSpec {
  Display *common.Display
  Plugins []Plugin
}

type Plugin struct {
  Kind Kind
  Metadata PluginMetadata
  Spec PluginSpec
}
```

AntoineThebaud commented on internal/api/plugin/npm.go:
```diff
@@ -39,14 +39,14 @@ type BuildInfo struct {
 }
 
-type Metadata struct {
+type ManifetsMetadata struct {
```
```suggestion
type ManifestMetadata struct {
```

jgbernalp commented:
I adjusted the plugin module structure to match better our types conventions as discussed. I added the changes from the plugins side here: https://github.com/perses/plugins/pull/18

jgbernalp commented:
yes exactly. I'll merge it then. Thnx for checking.
